### ISSUE TITLE
How can I train model on machine translation task?

### ISSUE THREAD

**Issue by blackredscarf** [facebookresearch/ParlAI#1255]
I intend to add my model to train on wmt task. But I found  ParlAi can only build one dictionary, and only support English tokenize. 

Could you tell me how can I do?

**Comment by stephenroller** [comment]
You can use re or bpe tokenization; those are both language agnostic. I believe it is `-t iwslt` to do English German translation. Indeed, the same dictionary is going to be used for both languages, but that shouldn’t be a problem

**Comment by blackredscarf** [comment]
@stephenroller I know the same dictionary is available to be used for both languages, but I need to know the source language vocabulary size and the target language vocabulary size, and compute softmax, also index is a problem.

**Comment by stephenroller** [comment]
ParlAI doesn’t support duel dictionaries at this time. I don’t think we have plans to add support anytime soon.

You might find pytorch/fairseq is a better project choice for you, if you’re interested in doing specific NMT research.

**Comment by blackredscarf** [comment]
Thank you for your reply. I prefer ParlAI compared to fairseq because I can easily add models and take advantage of multitask training.

**Comment by stephenroller** [comment]
That's nice to hear!

If you wanted to hack it in, you could create a `TorchAgent` and override the `vectorize` to handle the separate vectorization of English and German. It would be a bit dirty, but you could do it.

**Comment by alexholdenmiller** [comment]
You can also extend the dictionary class to make `BilingualDictAgent(DictionaryAgent)` (see the DrQA agent's extended dictionary below for an example of extending the dictionary), which could have another tokenizer and which could vectorize the text (one language) differently from the labels (another language).

https://github.com/facebookresearch/ParlAI/blob/master/parlai/agents/drqa/drqa.py#L46-L88

https://github.com/facebookresearch/ParlAI/blob/master/parlai/agents/drqa/drqa.py#L103-L105
